geomesa-compute/src/main/scala/org/locationtech/geomesa/compute/spark/GeoMesaSpark.scala

```diff
@@ -68,20 +87,43 @@ object GeoMesaSpark {
 
     ConfiguratorBase.setConnectorInfo(classOf[AccumuloInputFormat], conf, ds.connector.whoami(), ds.authToken)
 
-    if(useMock) ConfiguratorBase.setMockInstance(classOf[AccumuloInputFormat], conf, ds.connector.getInstance().getInstanceName)
-    else ConfiguratorBase.setZooKeeperInstance(classOf[AccumuloInputFormat], conf, ds.connector.getInstance().getInstanceName, ds.connector.getInstance().getZooKeepers)
-
+    if (useMock){
+      ConfiguratorBase.setMockInstance(classOf[AccumuloInputFormat],
+        conf,
+        ds.connector.getInstance().getInstanceName)
+    } else {
+      ConfiguratorBase.setZooKeeperInstance(classOf[AccumuloInputFormat],
+        conf,
+        ds.connector.getInstance().getInstanceName,
+        ds.connector.getInstance().getZooKeepers)
+    }
     InputConfigurator.setInputTableName(classOf[AccumuloInputFormat], conf, ds.getSpatioTemporalTable(sft))
     InputConfigurator.setRanges(classOf[AccumuloInputFormat], conf, qp.ranges)
-    qp.iterators.foreach { is => InputConfigurator.addIterator(classOf[AccumuloInputFormat], conf, is) }
+    qp.iterators.foreach { is => InputConfigurator.addIterator(classOf[AccumuloInputFormat], conf, is)}
 
-    val rdd = sc.newAPIHadoopRDD(conf, classOf[AccumuloInputFormat], classOf[Key], classOf[Value])
+    if (!qp.columnFamilies.isEmpty) {
+      InputConfigurator.fetchColumns(classOf[AccumuloInputFormat],
+        conf,
+        qp.columnFamilies.map(cf => new AccPair[Text, Text](cf, null)))
+    }
 
-    rdd.mapPartitions { iter =>
-      val sft = SimpleFeatureTypes.createType(typeName, spec)
-      val decoder = SimpleFeatureDecoder(sft, featureEncoding)
-      iter.map { case (k: Key, v: Value) => decoder.decode(v.get()) }
+    if (numberOfSplits.isDefined) {
+      GeoMesaInputFormat.configureSplits(conf,
+        numberOfSplits.get * sc.getExecutorStorageStatus.length)
+      InputConfigurator.setAutoAdjustRanges(classOf[AccumuloInputFormat], conf, false)
+      InputConfigurator.setAutoAdjustRanges(classOf[GeoMesaInputFormat], conf, false)
+    }
```

Comment:
While that's more idiomatic, I feel like it makes it less obvious that the option should contain only a single value. Why is this preferred?